imagegw/shifter_imagegw/dockerv2.py

```diff
@@ -90,30 +102,62 @@ def _verify_manifest_signature(manifest, text, digest):
 
 def _setup_http_conn(url, cacert=None):
     """Prepare http connection object and return it."""
-    (protocol, url) = url.split('://', 1)
+    target = urlparse.urlparse(url)
     conn = None
-    if protocol == 'http':
-        port = 80
-    else:
-        port = 443
-    if url.find('/') >= 0:
-        (server, _) = url.split('/', 1)
-    else:
-        server = url
-    if ':' in server:
-        (server, portstr) = server.split(':', 1)
-        port = int(portstr)
-    if protocol == 'http':
-        conn = httplib.HTTPConnection(server, port=port)
-    elif protocol == 'https':
+
+    if target.scheme == 'http':
+        if 'http_proxy' in os.environ and needProxy(target.hostname):
```

Comment:
Indeed, I missed that one when I updated the code to match the new coding style.